### ISSUE TITLE
Missed semicolon has incorrect collumn 

### ISSUE THREAD

**Issue by Gusted** [stylelint/stylelint#5419]
> Clearly describe the bug

The `Missed semicolon` error will produce an incorrect column.

> Which rule, if any, is the bug related to?

`CssSyntaxError`

> What code is needed to reproduce the bug?

e.g.

```css
div {
        color: red !important
        background-color: #000
}
```

> What stylelint configuration is needed to reproduce the bug?

CSSSyntaxError is enabled by default and not a rule.

```json
{
  "rules": {}
}
```

> Which version of stylelint are you using?

13.13.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Demo, otherwise you can do it with ` stylelint test.css --config test.json`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Nope.

> What did you expect to happen?

Show the error:
```shell
test.css
 2:30  ✖  Missed semicolon   CssSyntaxError
```

With the column at `30` which is after `!important` and the correct place to add the semicolon

> What actually happened (e.g. what warnings or errors did you get)?

Show the error:
```shell
test.css
 2:20  ✖  Missed semicolon   CssSyntaxError
```

With the the column at `20` which is before `!important` and not the correct place to add the semicolon.
It should instead error at the correct place which is after `!important` and in this case at column `30`.

**Comment by ybiquitous** [comment]
Hi @Gusted, thank you for using the issue template.

I can reproduce the error on our [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIoEsBuACYAdAdu-B+kANhAE7zpkxToCEyAtgA7kAuAhjm7oYQEYcwAawDmZCAFccUALQlylAMQAGNbgC+uEABoQAZwCe3DgA8E4fft3gIOAGbJRF7HnRYQZScRj6PlYC0cIJANIA).

The cause of this error and inaccurate column is that PostCSS cannot parse such a CSS code. See below:

https://github.com/postcss/postcss/blob/1545d8aabd2a920c76a70fdd4e7c43a9464fba5e/lib/parser.js#L555

```console
$ cat a.css
div {
        color: red !important
        background-color: #000
}

$ node -pe 'require("postcss").parse(fs.readFileSync("a.css", "utf-8"))'
/Users/masafumi.koba/git/stylelint/stylelint/node_modules/postcss/lib/parse.js:33
    throw e
    ^

CssSyntaxError: <css input>:2:20: Missed semicolon
    at Input.error (/Users/masafumi.koba/git/stylelint/stylelint/node_modules/postcss/lib/input.js:123:16)
    at Parser.checkMissedSemicolon (/Users/masafumi.koba/git/stylelint/stylelint/node_modules/postcss/lib/parser.js:555:22)
    at Parser.decl (/Users/masafumi.koba/git/stylelint/stylelint/node_modules/postcss/lib/parser.js:253:12)
    at Parser.other (/Users/masafumi.koba/git/stylelint/stylelint/node_modules/postcss/lib/parser.js:147:12)
    at Parser.parse (/Users/masafumi.koba/git/stylelint/stylelint/node_modules/postcss/lib/parser.js:59:16)
    at Function.parse (/Users/masafumi.koba/git/stylelint/stylelint/node_modules/postcss/lib/parse.js:11:12)
    at [eval]:1:20
    at Script.runInThisContext (node:vm:129:12)
    at Object.runInThisContext (node:vm:305:38)
    at node:internal/process/execution:81:19 {
  reason: 'Missed semicolon',
  source: 'div {\n        color: red !important\n        background-color: #000\n}\n',
  line: 2,
  column: 20,
  input: {
    line: 2,
    column: 20,
    source: 'div {\n        color: red !important\n        background-color: #000\n}\n'
  }
}
```

**Comment by Gusted** [comment]
Hi @ybiquitous,

Thank you for responding.
Does that mean postCSS is the one giving inaccurate column data in this case, if so, should the issue being taken over to their repo?

**Comment by ybiquitous** [comment]
@Gusted Sorry, I’m not familiar with the details of the PostCSS parser, but I think you may get more help if you open an issue on the PostCSS repository.

**Comment by alexander-akait** [comment]
It is not CSS/LESS/SCSS/SASS/etc, invalid CSS === linter doesn't work

**Comment by alexander-akait** [comment]
And even more, there is not problem with built-in CSS parser in postcss, you need custom parser for this weird CSS

**Comment by Gusted** [comment]
> It is not CSS/LESS/SCSS/SASS/etc, invalid CSS === linter doesn't work

That's correct, but the error in this one comes from a Syntaxerror, which should error at the correct place IMO. It's a bit weird to go error at some place which is incorrect at it's own.



> you need custom parser for this weird CSS

Not sure how this is `weird css` I actually found about this bug when I forgot to add my semi-column and stylelint was erroring it should add an semi-column at the wrong place.


Anyway, thanks for diagnosing the bug - hopefully can be handled over at postcss.

**Comment by ybiquitous** [comment]
@Gusted Great! The PR you created has been merged soon! 👏🏼
